javascriptv3/example_code/ssm/tests/create-maintenance-window.unit.test.js

```diff
@@ -0,0 +1,92 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { CreateMaintenanceWindowCommand, SSMClient } from "@aws-sdk/client-ssm";
+import { main } from "../actions/create-maintenance-window.js";
+
+describe("createMaintenanceWindow", () => {
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
+  it("should create a new maintenance window successfully", async () => {
+    const mockWindowId = "test-window-id";
+    const sendMock = vi
+      .spyOn(SSMClient.prototype, "send")
```

Comment:
Spying on the prototype is actually something I hadn't thought of.

Reply:
Any issue with doing it this way?

Reply:
I'd have to test it, but the SSMClient import would be shared with other objects in the runtime, and there's a potential of the mock bleeding into other tests. But, I think vitest handles this.

Reply:
Played around with it and verified here: https://vitest.dev/guide/improving-performance.html#test-isolation